### PR TITLE
Add stable api version 2023-12-01 for ScheduledQueryRules

### PR DESCRIPTION
**Pull request by avjerafi**
# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

File: specification/monitor/resource-manager/readme.md
Comment (kazrael2119) [inline specification/monitor/resource-manager/readme.md:47]: could you add these two in this api version？
``` 
- Microsoft.Monitor/stable/2023-04-03/monitoringAccounts_API.json
- Microsoft.Monitor/stable/2023-04-03/operations_API.json
```

Comment (avjerafi) [reply specification/monitor/resource-manager/readme.md:47]: @kazrael2119 Sure. Done.

**Comment by avjerafi** [comment]
@kazrael2119 Regarding the LintDiff check: This api version introduces no schema changes. The "additionalProperties" is already in use in earlier api versions. I cannot stop using the "additionalProperties" now without introducing a major breaking change. What do I need to do in order to resolve this case?

File: specification/monitor/resource-manager/readme.md
Comment (kazrael2119) [reply specification/monitor/resource-manager/readme.md:47]: @avjerafi ,sorry , I missed all the other resourceTypes under "package-2023-10", please add below too
```
- Microsoft.Insights/stable/2022-10-01/autoscale_API.json
  - Microsoft.Insights/stable/2015-04-01/operations_API.json
  - Microsoft.Insights/stable/2016-03-01/alertRulesIncidents_API.json
  - Microsoft.Insights/stable/2016-03-01/alertRules_API.json
  - Microsoft.Insights/stable/2016-03-01/logProfiles_API.json
  - Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettings_API.json
  - Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettingsCategories_API.json
  - Microsoft.Insights/stable/2023-01-01/actionGroups_API.json
  - Microsoft.Insights/preview/2023-05-01-preview/tenantActionGroups_API.json
  - Microsoft.Insights/stable/2015-04-01/activityLogs_API.json
  - Microsoft.Insights/stable/2015-04-01/eventCategories_API.json
  - Microsoft.Insights/stable/2015-04-01/tenantActivityLogs_API.json
  - Microsoft.Insights/stable/2023-10-01/metricDefinitions_API.json
  - Microsoft.Insights/stable/2023-10-01/metrics_API.json
  - Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json
  - Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
  - Microsoft.Insights/preview/2022-08-01-preview/scheduledQueryRule_API.json
  - Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
  - Microsoft.Insights/preview/2018-11-27-preview/vmInsightsOnboarding_API.json
  - Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json
  - Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json
  - Microsoft.Insights/stable/2022-06-01/dataCollectionEndpoints_API.json
  - Microsoft.Insights/stable/2022-06-01/dataCollectionRuleAssociations_API.json
  - Microsoft.Insights/stable/2022-06-01/dataCollectionRules_API.json
  ```

**Comment by kazrael2119** [comment]
@avjerafi ,could you fix the duplicate schema ErrorContract, in your resource type , it's a manually definition, but in 
image
It references from ```common-types/v2/commonMonitoringTypes.json#/definitions/ErrorContract```

**Comment by avjerafi** [comment]
@kazrael2119 I moved to use the ErrorContract from common-types as you requested. Even though I made sure the schema for both types is completely identical, it resulted in an unjustified flagging as breaking change:
https://github.com/Azure/azure-rest-api-specs/pull/27354/checks?check_run_id=20715295975

How can this be suppressed?

**Comment by kazrael2119** [comment]
> @kazrael2119 I moved to use the ErrorContract from common-types as you requested. Even though I made sure the schema for both types is completely identical, it resulted in an unjustified flagging as breaking change: https://github.com/Azure/azure-rest-api-specs/pull/27354/checks?check_run_id=20715295975
> 
> How can this be suppressed?

I think you should ask BreaingChangeReviewBoard to get approval

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json
Comment (mikekistler) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json:723]: Please delete this line -- I think this is what is causing the breaking changes tooling to complain.
```suggestion
```

Comment (avjerafi) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json]: Thank you @mikekistler. I actually just pushed a fix right now, which references the common-types "ErrorContract" directly inside the "schema" field, instead of going through the locally defined wrapper "ErrorContract". I hope this will resolve it.

**Comment by avjerafi** [comment]
> @rkmanda Regarding the LintDiff check: This api version introduces no schema changes. The "additionalProperties" is already in use in earlier api versions. I cannot stop using the "additionalProperties" now without introducing a major breaking change. What do I need to do in order to resolve this case?

@rkmanda Gentle ping on this

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json
Comment (visingla-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json:331]: Since actionProperties is being added to "stable" API version first time, that's why this linter error is asking about not to use additionalProperties.

A few questions:
1. I don't see any examples getting updated which show the usage of "actionProperties". Please update the examples to show its usage.
2. Why is there a need to a dictionary here? Why can't it be a list of properties, where each object in the array list can be "Property Name" and "Property Value"? That way, you can validate the property name if it is valid or not.
3. How is this different from customProperties?

Comment (avjerafi) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json:331]: 1. Done.
2. This section in the payload is populated with key-value pairs which are passed as-is without validation on our part, via our service's data plane, and are eventually interpreted by a service several hops down the pipe. We do not control the allowed values and we should not validate their content.
3. customProperties are similarly passed as-is, but are strictly for the customer's use. They are added to the notification payload of the alert and are there for the user to categorize and enrich their alerts. They carry no meaning to any service, and therefore are separated from actionProperties.

We explained our scenario in an API Modeling Work Hours session with @rkmanda and he gave us approval for a suppression for the actionProperties field.

**Comment by avjerafi** [comment]
@rkmanda I've pushed a suppression for the path of the actionProperties field, but saw that the flagging of the fields which refer to it did not go away, even after I pushed an additional commit to add their paths to the suppression - which I did not think made much sense. Are these just false positives by LintDiff?

When I saw that it did not help I reverted that last commit. The only current suppression is for the actionProperties field itself.

**Review by rkmanda — APPROVED** [review]
:shipit:

**Comment by kazrael2119** [comment]
add @JeffreyRichter as a reviewer because the default tag has multi api versions

**Comment by JeffreyRichter** [comment]
Azure's policy requires that all the api-versions match uniformly.  So, you can't have some swaggers from one version and some from another. You also cannot have some be release and some be preview. There is no way a customer can rationalize how to use the service in this case. We are working on adding tooling to the repo to automatically catch these violations but we have to do it manually for now.

**Comment by avjerafi** [comment]
@kazrael2119 Can you please clarify the current status? How do we resolve the issue of multiple api versions in the default tag? What actions must I take?

**Comment by avjerafi** [comment]
@rkmanda The PR is still flagged by Go and JS breaking change tags for weeks after ARM signed off. IIUC, the workflow diagram basically says "wait", but I don't see any reviewer addressing these. Can you please let me know what I must do to expedite the resolution?

**Comment by jhendrixMSFT** [comment]
@lirenhe @Alancere can somebody please review the Go breaking changes?

**Review by ms-henglu — APPROVED** [review]

**Pull request merged**
